pkg/crc/pullsecret/pullsecret.go

```diff
@@ -0,0 +1,169 @@
+	output, err := drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | tee /tmp/pull-secret.yaml\n%s\nEOF", fmt.Sprintf(pullSecret, base64OfPullSec)))
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output is : %s", output)
+	return nil
+}
+
+func addpullSecretToIntanceDisk(driver drivers.Driver, pullSec string) error {
+	output, err := drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | sudo tee /var/lib/kubelet/config.json\n%s\nEOF", pullSec))
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output is : %s", output)
+	return nil
+}
+
+func addKubeconfigFileToInstance(driver drivers.Driver, kubeconfigFilePath string) error {
+	kubeconfig, err := ioutil.ReadFile(kubeconfigFilePath)
+	if err != nil {
+		return err
+	}
+	_, err = drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | tee /tmp/kubeconfig\n%s\nEOF", string(kubeconfig)))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func replaceUserPullSecret(driver drivers.Driver) error {
+	output, err := drivers.RunSSHCommandFromDriver(driver, replacePullSecretCmd)
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output of %s: %s", replacePullSecretCmd, output)
+	return nil
+}
+
+func addClusterID(driver drivers.Driver) error {
+	clusterID := uuid.New()
+	output, err := drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf(updateClusterIdCmd, clusterID))
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output of %s: %s", updateClusterIdCmd, output)
+	return nil
+}
+
+func setPullSecretAndClusterID(driver drivers.Driver) (rerr error) {
+	sd := systemd.NewInstanceSystemdCommander(driver)
+	defer func() {
+		// Stop the kublet service.
+		if _, err := sd.Stop("kubelet"); err != nil {
+			rerr = err
+		}
+		output, err := drivers.RunSSHCommandFromDriver(driver, stopAndRemovePodsCmd)
+		logging.DebugF("Output of %s: %s", stopAndRemovePodsCmd, output)
+		if err != nil {
+			rerr = err
+		}
+	}()
+
+	// Start the kubelet service
+	if _, err := sd.Start("kubelet"); err != nil {
+		return err
+	}
+
+	// Replace existing pull secret to the user pull secret
+	if err := replaceUserPullSecret(driver); err != nil {
+		return err
+	}
+
+	// add random cluster id
+	if err := addClusterID(driver); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func StorePullSecretToCache(pullsecret string, destDir string, fileName string, mode os.FileMode) error {
+	err := os.MkdirAll(destDir, mode)
+	if err != nil && !os.IsExist(err) {
+		return fmt.Errorf("Cannot create directory %s: %v", destDir, err)
+	}
+	pullSecretFilePath := filepath.Join(destDir, fileName)
+	if err := ioutil.WriteFile(pullSecretFilePath, []byte(pullsecret), mode); err != nil {
+		return fmt.Errorf("Cannot write to %s: %v", pullSecretFilePath, err)
+	}
+	return nil
+}
+
+func GetPullSecretFromCache(pullSecretFilePath string) (string, error, bool) {
+	// Check if file exist
+	if _, err := os.Stat(pullSecretFilePath); os.IsNotExist(err) {
+		return "", nil, false
+	}
```

Comment:
Isn't this ValidatePath(pullSecretFilePath) ?

Reply:
@cfergeau this create cyclic dependency which is not allowed in go :(.

```
package github.com/code-ready/crc/cmd/crc
	imports github.com/code-ready/crc/cmd/crc/cmd
	imports github.com/code-ready/crc/cmd/crc/cmd/config
	imports github.com/code-ready/crc/pkg/crc/config
	imports github.com/code-ready/crc/pkg/crc/validation
	imports github.com/code-ready/crc/pkg/crc/machine
	imports github.com/code-ready/crc/pkg/crc/pullsecret
	imports github.com/code-ready/crc/pkg/crc/validation
```

Reply:
Ok :(